### PR TITLE
[rules/libyang.mk]: Fix the dependencies among libyang PKGs.

### PR DESCRIPTION
**Pull request by praveen-li**
This changes is needed to support parallel build Jobs.
Signed-off-by: Praveen Chaudhary pchaudhary@linkedin.com

**- Why I did it**

Fix the dependencies among libyang PKGs.

**- How I did it**
Made LIBYANG_PY2 and LIBYANG_PY3 depend on LIBYANG and LIBYANG_CPP. Also LIBYANG_CPP depends on LIBYANG.

**- How to verify it**

Have to depend on Jenkins Job, since problem is seen with (e.g., SONIC_BUILD_JOBS=20).

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by praveen-li** [comment]
@zhenggen-xu @jleveque : Kindly review. Thx.

File: rules/libyang.mk [hunk 1]
```diff
@@ -20,12 +20,15 @@ LIBYANG_DBG = libyang-dbg_$(LIBYANG_VERSION)_$(CONFIGURED_ARCH).deb
 $(eval $(call add_derived_package,$(LIBYANG),$(LIBYANG_DBG)))
 
 LIBYANG_CPP = libyang-cpp_$(LIBYANG_VERSION)_$(CONFIGURED_ARCH).deb
+$(LIBYANG_CPP)_DEPENDS +=  $(LIBYANG)
```
Comment (jleveque) [inline rules/libyang.mk:23]: Remove extra space after `+=`

File: rules/libyang.mk [hunk 2]
```diff
@@ -20,12 +20,15 @@ LIBYANG_DBG = libyang-dbg_$(LIBYANG_VERSION)_$(CONFIGURED_ARCH).deb
...
 $(eval $(call add_derived_package,$(LIBYANG),$(LIBYANG_CPP)))
 
 LIBYANG_PY3 = python3-yang_$(LIBYANG_VERSION)_$(CONFIGURED_ARCH).deb
+$(LIBYANG_PY3)_DEPENDS +=  $(LIBYANG) $(LIBYANG_CPP)
```
Comment (jleveque) [inline rules/libyang.mk:27]: Remove extra space after `+=`

File: rules/libyang.mk [hunk 3]
```diff
@@ -20,12 +20,15 @@ LIBYANG_DBG = libyang-dbg_$(LIBYANG_VERSION)_$(CONFIGURED_ARCH).deb
...
 $(eval $(call add_derived_package,$(LIBYANG),$(LIBYANG_PY3)))
 
 LIBYANG_PY2 = python2-yang_$(LIBYANG_VERSION)_$(CONFIGURED_ARCH).deb
+$(LIBYANG_PY2)_DEPENDS +=  $(LIBYANG) $(LIBYANG_CPP)
```
Comment (jleveque) [inline rules/libyang.mk:31]: Remove extra space after `+=`

**Comment by praveen-li** [comment]
Thx for the test run,
I fixed whitespace issues. Thx.




Thanks,

Regards
Praveen
________________________________

**Comment by jleveque** [comment]
My first test build (with <= 20 build jobs) did not encounter any dependency issues. I think this PR fixes the issue. 👍

**Comment by jleveque** [comment]
Retest mellanox please

**Comment by praveen-li** [comment]
> My first test build (with <= 20 build jobs) did not encounter any dependency issues. I think this PR fixes the issue. 👍

Cool, Then. Happy Weekend.

**Pull request merged**